File: frontend/lib/market-config/market-config.ts
// Central market configuration for the current default market (Colombia / Hapi).
// Import these values instead of hard-coding currency codes, pairs, or provider
// names so new countries and brokers can be added without touching every file.

export const MARKET_CONFIG = {
  // Country / broker defaults for onboarding and presets.
  defaultCountry: "co",
  defaultBrokerId: "hapi-colombia",

  // Currencies.
  baseCurrency: "USD",
  localCurrency: "COP",
  baseCurrencyLabel: "US Dollar",
  localCurrencyLabel: "Colombian Peso",
  defaultCurrencyPair: "USD/COP",
  inverseCurrencyPair: "COP/USD",

  // Cash-flow currencies allowed in filters and forms.
  cashFlowCurrencies: ["USD", "COP"] as const,

  // Market data / FX provider defaults.
  priceProvider: "twelve-data",
  defaultMarketCurrency: "USD",
  defaultFxRateSource: "twelve-data",
  defaultFxRateDays: 30,
  maxFxRateDays: 90,

  // Display defaults.
  baseCurrencyDecimals: 2,
  localCurrencyDecimals: 0,
} as const

export type CashFlowCurrency = (typeof MARKET_CONFIG.cashFlowCurrencies)[number]

export function isCashFlowCurrency(value: string): value is CashFlowCurrency {
  return MARKET_CONFIG.cashFlowCurrencies.includes(value as CashFlowCurrency)
}

export function formatCurrencyPair(from: string, to: string): string {
  return `${from}/${to}`
}

File: backend/internal/config/market_config.go
// Package config holds market-specific defaults shared across the backend.
// Centralizing these values makes it possible to add new countries, currencies,
// and brokers without scattering literals through handlers and services.
package config

import "time"

// Currency codes for the current default market (Colombia / Hapi).
const (
	BaseCurrency   = "USD"
	LocalCurrency  = "COP"
	DefaultCurrencyPair = BaseCurrency + "/" + LocalCurrency
	InverseCurrencyPair = LocalCurrency + "/" + BaseCurrency
)

// Market-data provider defaults.
const (
	TwelveDataSource    = "twelve-data"
	TwelveDataBaseURL   = "https://api.twelvedata.com"
	DefaultMarketCurrency = BaseCurrency
)

// Cache and chart defaults.
const (
	MarketDataCacheTTL    = 24 * time.Hour
	MarketPriceCooldown   = 60 * time.Second
	DefaultFXRateDays     = 30
	MaxFXRateDays         = 90
)

// BatchAPIBatchSymbols is the maximum number of symbols sent in a single Twelve Data
// batch request. Twelve Data does not document a hard cap; 20 keeps requests small
// and avoids URL length limits.
const MaxBatchSymbols = 20

// SupportedCurrencyPairs are the currency pairs the current FX handlers support.
var SupportedCurrencyPairs = []string{DefaultCurrencyPair, InverseCurrencyPair}
